egs/callhome_diarization/v1/run.sh

```diff
@@ -0,0 +1,247 @@
+      --cmd "$train_cmd" --write-utt2num-frames true \
+      data/$name exp/make_mfcc $mfccdir
+    utils/fix_data_dir.sh data/$name
+  done
+
+  for name in train callhome1 callhome2; do
+    sid/compute_vad_decision.sh --nj 40 --cmd "$train_cmd" \
+      data/$name exp/make_vad $vaddir
+    utils/fix_data_dir.sh data/$name
+  done
+
+  # The sre dataset is a subset of train
+  cp data/train/{feats,vad}.scp data/sre/
+  utils/fix_data_dir.sh data/sre
+
+  # Create segments for ivector extraction for PLDA training data.
+  echo "0.01" > data/sre/frame_shift
+  diarization/vad_to_segments.sh --nj 40 --cmd "$train_cmd" \
+    data/sre data/sre_segmented
+fi
+
+# Train UBM and i-vector extractor
+if [ $stage -le 2 ]; then
+  # Reduce the amount of training data for the UBM.
+  utils/subset_data_dir.sh data/train 16000 data/train_16k
+  utils/subset_data_dir.sh data/train 32000 data/train_32k
+
+  # Train UBM and i-vector extractor.
+  sid/train_diag_ubm.sh --cmd "$train_cmd --mem 20G" \
+    --nj 40 --num-threads 8 --delta-order 1 --apply-cmn false \
+    data/train_16k $num_components exp/diag_ubm_$num_components
+
+  sid/train_full_ubm.sh --nj 40 --remove-low-count-gaussians false \
+    --cmd "$train_cmd --mem 25G" --apply-cmn false \
+    data/train_32k exp/diag_ubm_$num_components \
+    exp/full_ubm_$num_components
+
+  sid/train_ivector_extractor.sh \
+    --cmd "$train_cmd --mem 35G" \
+    --ivector-dim $ivector_dim --num-iters 5 --apply-cmn false \
+    exp/full_ubm_$num_components/final.ubm data/train \
+    exp/extractor_c${num_components}_i${ivector_dim}
+fi
+
+# Extract i-vectors
+if [ $stage -le 3 ]; then
+  # Reduce the amount of training data for the PLDA,
+  utils/subset_data_dir.sh data/sre_segmented 128000 data/sre_segmented_128k
+  # Extract iVectors for the SRE, which is our PLDA training
+  # data.  A long period is used here so that we don't compute too
+  # many iVectors for each recording.
+  diarization/extract_ivectors.sh --cmd "$train_cmd --mem 25G" \
+    --nj 40 --window 3.0 --period 10.0 --min-segment 1.5 --apply-cmn false \
+    --hard-min true exp/extractor_c${num_components}_i${ivector_dim} \
+    data/sre_segmented_128k exp/ivectors_sre_segmented_128k
+
+  # Extract iVectors for the two partitions of callhome.
+  diarization/extract_ivectors.sh --cmd "$train_cmd --mem 20G" \
+    --nj 40 --window 1.5 --period 0.75 --apply-cmn false \
+    --min-segment 0.5 exp/extractor_c${num_components}_i${ivector_dim} \
+    data/callhome1 exp/ivectors_callhome1
+
+  diarization/extract_ivectors.sh --cmd "$train_cmd --mem 20G" \
+    --nj 40 --window 1.5 --period 0.75 --apply-cmn false \
+    --min-segment 0.5 exp/extractor_c${num_components}_i${ivector_dim} \
+    data/callhome2 exp/ivectors_callhome2
+fi
+
+# Train PLDA models
+if [ $stage -le 4 ]; then
+  # Train a PLDA model on SRE, using callhome1 to whiten.
+  # We will later use this to score iVectors in callhome2.
+  run.pl exp/ivectors_callhome1/log/plda.log \
```

Comment:
Is there a reason why this is run.pl? I think you should use $train_cmd.

Reply:
I can get behind that, although I don't think it matters much as the computation is very fast.